client/Components/map.jsx

```diff
@@ -49,6 +55,38 @@ const Map = (props) => {
     getEggs();
   }, []);
 
+  const toggleEggDetails = (event, data) => {
+    const egg = event.target.getAttribute("value");
+    if (!egg) setTargetEgg(null);
+    if (egg === "egg") {
```

Comment:
I don't really undertand what this is. What could `egg` be instead of `"egg"`? Usually we use `data-` attributes for custom attributes instead of value. 

Also, if possible it's better to have a guard with a `return` so that all of your logic doesn't go inside of a big `if` block.